### ISSUE TITLE
Bug on related_models_with_context with single resource

### ISSUE THREAD

**Issue by thibaudgg** [venuu/jsonapi-authorization#62]
Hi @valscion,

While testing the v1.0.0.alpha1 release on our app, I got a failing test on the `AuthorizingProcessor#related_models_with_context` on line [280](https://github.com/venuu/jsonapi-authorization/blob/master/lib/jsonapi/authorization/authorizing_processor.rb#L280).

Basically if `assoc_value` is not an `Hash` or an `Array` then `assoc_value[:id]` should not be used, but rather `assoc_value` instead. Which would give (and make my test pass):

``` ruby
resource_class.find_by_key(assoc_value, context: context)._model
```

What do you think? Could you fix that one directly one master or should I submit a PR with a spec?

**Comment by valscion** [comment]
Oh yeah, this change was part of the diff in https://github.com/venuu/jsonapi-authorization/pull/60

I'll cherry-pick b4233b8e45c5371aee396fde5770c2271ef65708 to a new PR and merge + release a new version. Thanks!

**Comment by gnfisher** [comment]
I am aware of this, I need to push up the fix separately from issue I'm currently working on. Will do so today!

**Comment by valscion** [comment]
On another note, it'd be awesome to have a test that would've failed because of this — I assume some specs in #60 might've failed without this fix..?

**Comment by gnfisher** [comment]
Yes, I discovered it while working on specs in #60. I will include specs that test it in that PR.

**Comment by valscion** [comment]
Released in v1.0.0.alpha2 :)
